helix-core/src/test/java/org/apache/helix/task/TestDynamicTaskLoading.java

```diff
@@ -0,0 +1,251 @@
+import org.testng.annotations.Test;
+
+public class TestDynamicTaskLoading extends ZkTestBase {
+  private final String _clusterName = CLUSTER_PREFIX + "_" + getShortClassName();
+  private static final String _instanceName = "localhost_12913";
+  private HelixManager _manager;
+  private MockParticipantManager _participant;
+  private ClusterControllerManager _controller;
+
+  @BeforeMethod
+  public void beforeMethod() throws Exception {
+    _gSetupTool.addCluster(_clusterName, true);
+
+    _manager = HelixManagerFactory
+        .getZKHelixManager(_clusterName, "Admin", InstanceType.ADMINISTRATOR, ZK_ADDR);
+    _manager.connect();
+
+    _gSetupTool.addInstanceToCluster(_clusterName, _instanceName);
+    _participant = new MockParticipantManager(ZK_ADDR, _clusterName, _instanceName);
+    StateModelFactory<StateModel> stateModelFactory =
+        new MasterSlaveStateModelFactory(_instanceName, 0);
+    StateMachineEngine stateMach = _participant.getStateMachineEngine();
+    stateMach.registerStateModelFactory("MasterSlave", stateModelFactory);
+    Map<String, TaskFactory> taskFactoryReg = new HashMap<>();
+    _participant.getStateMachineEngine().registerStateModelFactory(TaskConstants.STATE_MODEL_NAME,
+        new TaskStateModelFactory(_participant, taskFactoryReg));
+    _participant.syncStart();
+
+    _controller = new ClusterControllerManager(ZK_ADDR, _clusterName, null);
+    _controller.syncStart();
+  }
+
+  private void removePathIfExists(String path) {
+    if (_manager.getHelixDataAccessor().getBaseDataAccessor().exists(path, 0)) {
+      _manager.getHelixDataAccessor().getBaseDataAccessor().remove(path, 0);
+    }
+  }
+
+  /**
+   * Submit a workflow consisting of a job with a MockTask task.
+   * @param workflowName name of the workflow
+   * @param driver {@link TaskDriver} to submit workflowName to
+   */
+  private void submitWorkflow(String workflowName, TaskDriver driver) {
+    JobConfig.Builder job = new JobConfig.Builder();
+    job.setJobCommandConfigMap(Collections.singletonMap(MockTask.JOB_DELAY, "100"));
+    Workflow.Builder workflow = new Workflow.Builder(workflowName);
+    job.setWorkflow(workflowName);
+    TaskConfig taskConfig =
+        new TaskConfig(MockTask.TASK_COMMAND, new HashMap<String, String>(), null, null);
+    job.addTaskConfigMap(Collections.singletonMap(taskConfig.getId(), taskConfig));
+    job.setJobId(TaskUtil.getNamespacedJobName(workflowName, "JOB"));
+    workflow.addJob("JOB", job);
+    driver.start(workflow.build());
+  }
+
+  @Test
+  public void testDynamicTaskLoading() throws Exception {
+    // Add task definition information as a DynamicTaskConfig.
+    List<String> taskClasses = new ArrayList<String>();
+    taskClasses.add("com.mycompany.mocktask.MockTask");
+    DynamicTaskConfig taskConfig =
+        new DynamicTaskConfig("Reindex", "src/test/resources/Reindex.jar", "1.0.0", taskClasses,
+            "com.mycompany.mocktask.MockTaskFactory");
```

Comment:
Pull out your strings and name them as a local variable - for example,

String fullyQualifiedTaskClassName = "com.mycompany.mocktask.MockTask"; 
...

for readability.